strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/layout/pypi/PypiArtifactController.java

```diff
@@ -0,0 +1,351 @@
+                                                                              repository.getStorage().getId(),
+                                                                              repository.getId(),
+                                                                              coordinates.toPath());
+
+        provideArtifactDownloadResponse(request, response, headers, repositoryPath);
+    }
+
+    @ApiOperation(value = "This Endpoint will be used to retreive all the versions of packages present in artifactory.")
+    @ApiResponses(value = { @ApiResponse(code = HttpURLConnection.HTTP_OK, message = "Success"),
+                            @ApiResponse(code = HttpURLConnection.HTTP_NOT_FOUND, message = "Request Url Not Found"),
+                            @ApiResponse(code = HttpURLConnection.HTTP_INTERNAL_ERROR, message = "An error occurred while executing download request."),
+                            @ApiResponse(code = HttpURLConnection.HTTP_UNAVAILABLE, message = "Service Unavailable.") })
+    @PreAuthorize("hasAuthority('ARTIFACTS_RESOLVE')")
+    @RequestMapping(path = "{storageId}/{repositoryId}/simple/{packageName}/", method = RequestMethod.GET, produces = MediaType.TEXT_HTML)
+    public ResponseEntity<String> getPackagePath(@RepositoryMapping Repository repository,
+                                                 @PathVariable(name = "packageName") String packageName,
+                                                 HttpServletRequest request,
+                                                 HttpServletResponse response,
+                                                 @RequestHeader HttpHeaders headers)
+        throws Exception
+    {
+
+        // https://www.python.org/dev/peps/pep-0427/#escaping-and-unicode
+        final String packageNameToDownload = packageName.replaceAll("[^A-Za-z0-9 ]", "_");
+
+        logger.info("Get package path request for storageId -> [{}] , repositoryId -> [{}], packageName -> [{}]",
+                    repository.getStorage().getId(),
+                    repository.getId(), packageNameToDownload);
+
+        RepositoryProvider repositoryProvider = repositoryProviderRegistry.getProvider(repository.getType());
+
+        Predicate predicate = Predicate.empty();
+        predicate.and(Predicate.of(ExpOperator.EQ.of("artifactCoordinates.coordinates.packaging",
+                                                     PypiArtifactCoordinates.WHEEL_EXTENSION)));
+        predicate.and(Predicate.of(ExpOperator.EQ.of("artifactCoordinates.coordinates.distribution",
+                                                     packageNameToDownload)));
+
+        Paginator paginator = new Paginator();
+        List<Path> searchResult = repositoryProvider.search(repository.getStorage().getId(), repository.getId(),
+                                                            predicate, paginator);
+
+        String searchPackageHtmlResponse = getHtmlResponse(searchResult, packageName, repository);
```

Comment:
does it the same as we have in `BrowseController`? I mean maybe response should be something like XML?

Reply:
I am not sure. Expected response is in html format.

Reply:
I mean does this method used by Pypi cli? 
This is just strange because cli tools normally requires xml or json.

Reply:
Yes it is used by pip internally.

Reply:
ok, thanks for info
just curious have you reverse engineered it or there is some docs?

Reply:
reverse engineering . analyzed traffic using wireshark